sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClient.java

```diff
@@ -0,0 +1,589 @@
+        }
+
+        ContainerRegistryImpl registryImpl = new ContainerRegistryImplBuilder()
+            .pipeline(httpPipeline)
+            .url(endpoint).buildClient();
+
+        this.endpoint = endpoint;
+        this.repositoryName = repositoryName;
+        this.registriesImplClient = registryImpl.getContainerRegistries();
+        this.serviceClient = registryImpl.getContainerRegistryRepositories();
+        this.apiVersion = apiVersion;
+    }
+
+    /**
+     * Get endpoint associated with the class.
+     * @return String the endpoint associated with this client.
+     */
+    public String getEndpoint() {
+        return this.endpoint;
+    }
+
+
+    /**
+     * Get the registry associated with the client.
+     * @return Return the registry name.
+     */
+    public String getRegistry() {
+        return this.endpoint;
+    }
+
+    /**
+     * Get repository associated with the class.
+     * @return Return the repository name.
+     * */
+    public String getRepository() {
+        return this.repositoryName;
+    }
+
+    /**
+     * Delete repository.
+     *
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DeleteRepositoryResult>> deleteWithResponse() {
+        return withContext(context -> deleteWithResponse(context));
+    }
+
+    /**
+     * Delete repository.
+     *
+     * @param context Context associated with the operation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the completion.
+     */
+    Mono<Response<DeleteRepositoryResult>> deleteWithResponse(Context context) {
+        try {
+            return this.registriesImplClient.deleteRepositoryWithResponseAsync(repositoryName, context)
+                .map(res -> Utils.mapResponse(res, Utils::mapDeleteRepositoryResult))
+                .onErrorMap(Utils::mapException);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Delete repository.
+     *
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @return the completion.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DeleteRepositoryResult> delete() {
+        return this.deleteWithResponse().flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Delete registry artifact.
+     *
+     * @param digest Digest name.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
```

Comment:
So, attempting to delete a non-existing registry artifact will result in ResourceNotFoundException whereas attempting to delete a non-existing repository will result in success? The javadoc for delete repository doesn't say that it throws ResourceNotFoundException. So, I am curious why these two delete operations have different behavior.

Reply:
Both result in a 404 - ResourceNotFoundException - I will go ahead and update things.

---
In reply to: [610944436](https://github.com/Azure/azure-sdk-for-java/pull/20368#discussion_r610944436) [](ancestors = 610944436)